src/hoverifier.ts

```diff
@@ -801,12 +823,134 @@ export function createHoverifier<C extends object, D, A>({
+                })),
+                // Do not emit anything after the code view this action came from got unhoverified
+                takeUntil(allUnhoverifies.pipe(filter(unhoverifiedCodeViewId => unhoverifiedCodeViewId === codeViewId)))
+            )
+        }),
+        share()
+    )
+
+    // Highlight the ranges returned by the document highlight provider
+    subscription.add(
+        documentHighlightObservables
+            .pipe(
+                switchMap(highlightObservable => highlightObservable),
+                switchMap(({ documentHighlightsOrError, position, adjustPosition, codeView, part, ...rest }) => {
+                    const highlights =
+                        documentHighlightsOrError &&
+                        documentHighlightsOrError !== LOADING &&
+                        !isErrorLike(documentHighlightsOrError)
+                            ? documentHighlightsOrError
+                            : []
+
+                    if (highlights.length === 0 || !position) {
+                        return of({ adjustPosition, codeView, part, ...rest, positions: of<Position[]>([]) })
+                    }
+
+                    const positions: Observable<Position>[] = []
+                    for (const { range } of highlights) {
+                        let pos = { ...position, ...range.start }
+
+                        // The requested position is is 0-indexed; the code here is currently 1-indexed
+                        const { line, character } = pos
+                        pos = { ...pos, line: line + 1, character: character + 1 }
+
+                        positions.push(
+                            adjustPosition
+                                ? from(
+                                      adjustPosition({
+                                          codeView,
+                                          direction: AdjustmentDirection.ActualToCodeView,
+                                          position: {
+                                              ...pos,
+                                              part,
+                                          },
+                                      })
+                                  )
+                                : of(pos)
+                        )
+                    }
+
+                    return of({
+                        adjustPosition,
+                        codeView,
+                        part,
+                        ...rest,
+                        positions: combineLatest(positions),
+                    })
+                }),
+                mergeMap(({ positions, codeView, dom, part }) =>
+                    positions.pipe(
+                        map(highlightedRanges =>
+                            highlightedRanges.map(highlightedRange =>
+                                highlightedRange
+                                    ? getTokenAtPosition(codeView, highlightedRange, dom, part, tokenize)
```

Comment:
This only highlights the token at the start of the range, right? Not full ranges? Do you plan to change that before merging? Otherwise it would be good to document that limitation somewhere